qos-service-impl/src/main/java/com/palantir/atlasdb/qos/config/SimpleThrottlingStrategy.java

```diff
@@ -20,11 +20,14 @@
 
 import com.palantir.atlasdb.qos.ratelimit.guava.RateLimiter;
 
-public class SimpleThrottlingStrategy implements ThrottlingStrategy {
+import net.jcip.annotations.GuardedBy;
+import net.jcip.annotations.ThreadSafe;
 
+@ThreadSafe
+public class SimpleThrottlingStrategy implements ThrottlingStrategy {
     private static final double ONCE_EVERY_HUNDRED_SECONDS = 0.01;
     private final RateLimiter rateLimiter;
-    private double multiplier;
+    @GuardedBy("this") private double multiplier;
```

Comment:
Should still be volatile, right?

Reply:
Sorry, doesn't need to also be volatile, but volatile might be good enough instead of synchronization - still guarded against concurrent access for now by the RateLimiter.

Reply:
Unfortunately, volatile will not provide the synchronization we need here:

```
As soon as a thread needs to first read the value of a volatile variable, and based on that value generate a new value for the shared volatile variable, a volatile variable is no longer enough to guarantee correct visibility. The short time gap in between the reading of the volatile variable and the writing of its new value, creates an race condition where multiple threads might read the same value of the volatile variable, generate a new value for the variable, and when writing the value back to main memory - overwrite each other's values.
```

Reply:
Ah yep, you're correct, since it's not atomic. Synchonized should be fine then.